prov/rxm/src/rxm_init.c

```diff
@@ -112,25 +113,28 @@ int rxm_info_to_rxm(uint32_t version, const struct fi_info *core_info,
 	info->caps = rxm_info.caps;
 	info->mode = core_info->mode | rxm_info.mode;
 
-	*info->tx_attr = *rxm_info.tx_attr;
-
-	info->tx_attr->msg_order = core_info->tx_attr->msg_order;
-	info->tx_attr->comp_order = core_info->tx_attr->comp_order;
-
+	info->tx_attr->caps		= rxm_info.tx_attr->caps;
+	info->tx_attr->mode		= info->mode;
+	info->tx_attr->msg_order 	= core_info->tx_attr->msg_order;
+	info->tx_attr->comp_order 	= core_info->tx_attr->comp_order;
+	info->tx_attr->inject_size	= rxm_info.tx_attr->inject_size;
 	/* Export TX queue size same as that of MSG provider as we post TX
 	 * operations directly */
-	info->tx_attr->size = core_info->tx_attr->size;
-
-	info->tx_attr->iov_limit = MIN(MIN(info->tx_attr->iov_limit,
-			core_info->tx_attr->iov_limit),
-			core_info->tx_attr->rma_iov_limit);
-
-	*info->rx_attr = *rxm_info.rx_attr;
-	info->rx_attr->iov_limit = MIN(info->rx_attr->iov_limit,
-			core_info->rx_attr->iov_limit);
-	/* Only SAS recv ordering can be guaranteed as RMA ops are not handled
-	 * by RxM protocol */
-	info->rx_attr->msg_order |= FI_ORDER_SAS;
+	info->tx_attr->size 		= core_info->tx_attr->size;
+	info->tx_attr->iov_limit 	= MIN(MIN(rxm_info.tx_attr->iov_limit,
+						  core_info->tx_attr->iov_limit),
+					      core_info->tx_attr->rma_iov_limit);
+	info->tx_attr->rma_iov_limit	= MIN(rxm_info.tx_attr->iov_limit,
+					      core_info->tx_attr->rma_iov_limit);
+
+	info->rx_attr->caps		= rxm_info.rx_attr->caps;
+	info->rx_attr->mode		= info->mode;
+	info->rx_attr->msg_order 	= core_info->rx_attr->msg_order;
+	info->rx_attr->comp_order 	= (core_info->rx_attr->comp_order |
+					   FI_ORDER_DATA);
```

Comment:
...and ORed back in here.  Why?

For large transfers, rxm uses RMA, which would require that the underlying provider support FI_ORDER_DATA in order for it to indicate that it is supported.  For smaller transfers I guess this is true, since you do memcpy, which I'm hoping copies data in order.

Reply:
FI_ORDER_DATA is defined as a completion order only for rx_attr. Currently there is no way to check if RMA read/write ops would follow ordering w.r.t memory.

> ...and ORed back in here. Why?

It was ORed because of the assumption that RxM can support the order bit by itself. It'll be removed since we can't.

Reply:
My question was why clear it only to reset it.  In any case, I'll wait for the update.

Reply:
The clearing part was for the hints passed to the underlying provider.